File: go/1760. Minimum Limit of Balls in a Bag.go
func minimumSize(nums []int, maxOperations int) int {
    left, right := 1, -1
    for _, i := range nums {
        right = max(right, i)
    }
    answer := -1
    for left <= right {
        mid := (left + right) / 2
        if isPossible(nums, mid, maxOperations) {
            answer = mid
            right = mid - 1
        } else {
            left = mid + 1
        }
    }
    return answer
}
func isPossible(nums []int, maxNum, maxOp int) bool {
    count := 0
    for _, i := range nums {
        if i <= maxNum {
            continue
        }
        count += i/maxNum
        if i % maxNum == 0 {
            count--
        }
    }
    return count <= maxOp
}

File: java/1760. Minimum Limit of Balls in a Bag.java
class Solution {
    public int minimumSize(int[] nums, int maxOperations) {
        int left = 1, right = -1;
        for (int i : nums) {
            right = Math.max(right, i);
        }
        int answer = -1;
        while (left <= right) {
            int mid = (left + right) / 2;
            if (isPossible(nums, maxOperations, mid)) {
                answer = mid;
                right = mid - 1;
            } else {
                left = mid + 1;
            }
        }
        return answer;
    }
    private boolean isPossible(int[] nums, int maxOp, int maxVal) {
        int count = 0;
        for (int i : nums) {
            if (i <= maxVal) continue;
            count += i/maxVal;
            if (i % maxVal == 0) count--;
        }
        return count <= maxOp;
    }
}
